aspnetcore/release-notes/aspnetcore-7.0.md

```diff
@@ -209,6 +209,309 @@ SignalR hub methods now support injecting services through dependency injection
+
+For more information, see the following sections of the *Virtualization* article:
+
+* [Root-level virtualization](xref:blazor/components/virtualization?view=aspnetcore-7.0#root-level-virtualization)
+* [Control the spacer element tag name](xref:blazor/components/virtualization?view=aspnetcore-7.0#control-the-spacer-element-tag-name)
+
+<!--
+
+    HOLD
+    
+    This is covered via the existing ref source link in the article.
+    I don't think we need to remark on it in the article, but you may wish to say something here.
+    I'll float some draft language.
+
+### `MouseEventArgs` updates
+
+`MovementX` and `MovementY` have been added to `MouseEventArgs`.
+
+For more information, see <xref:blazor/components/event-handling?view=aspnetcore-7.0#built-in-event-arguments>.
+
+-->
+
+### New Blazor loading page
+
+The Blazor WebAssembly project template has a new loading UI that shows the progress of loading the app.
+
+For more information, see <xref:blazor/fundamentals/startup?view=aspnetcore-7.0#loading-progress-indicators>.
+
+### Improved diagnostics for authentication in Blazor WebAssembly
+
+To help diagnose authentication issues in Blazor WebAssembly apps, detailed logging is available.
+
+For more information, see <xref:blazor/fundamentals/logging?view=aspnetcore-7.0#blazor-webassembly-authentication-logging>.
+
+<!--
+
+    IN PROGRESS
+
+### .NET JavaScript interop on WebAssembly
+
+.NET JavaScript `[JSImport]`/`[JSExport]` interop API is a new low-level mechanism for using .NET in Blazor WebAssembly and JavaScript-based apps. With this new JavaScript interop capability, you can invoke .NET code from JavaScript using the .NET WebAssembly runtime and call into JavaScript functionality from .NET without any dependency on the Blazor UI component model.
+
+For more information, see the following articles:
+
+* <xref:blazor/js-interop/import-export-interop>: Pertains only to Blazor WebAssebmly apps.
+* <xref:client-side/import-export-interop>: Pertains only to JavaScript apps that don't depend on the Blazor UI component model.
+
+Under consideration for docs ...
+
+[Blog Post (WebAssembly multithreading experimental)](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-7-rc-2/#webassembly-multithreading-experimental)
+
+-->
+
+### Conditional registration of the authentication state provider
+
+Prior to the release of .NET 7, `AuthenticationStateProvider` was registered in the service container with `AddScoped`. This made it difficult to debug apps, as it forced a specific order of service registrations when providing a custom implementation. Due to internal framework changes over time, it's no longer necessary to register `AuthenticationStateProvider` with `AddScoped`:
+
+```diff
+- builder.Services.AddScoped<AuthenticationStateProvider, ExternalAuthStateProvider>();
++ builder.Services.TryAddScoped<AuthenticationStateProvider, ExternalAuthStateProvider>();
+```
+
+### .NET WebAssembly build tools for .NET 6 projects
```

Comment:
I think this section might fit better in the migration section. Basically, if you're building libraries using the .NET WebAssembly build tools then to support .NET 6 & 7 then you need to multitarget and use the new wasm-tools-net6  workload.